meson.build

```diff
@@ -0,0 +1,228 @@
+  'include/freetype/ftsizes.h',
+  'include/freetype/ftsnames.h',
+  'include/freetype/ftstroke.h',
+  'include/freetype/ftsynth.h',
+  'include/freetype/ftsystem.h',
+  'include/freetype/fttrigon.h',
+  'include/freetype/fttypes.h',
+  'include/freetype/ftwinfnt.h',
+  'include/freetype/t1tables.h',
+  'include/freetype/ttnameid.h',
+  'include/freetype/tttables.h',
+  'include/freetype/tttags.h',
+]
+
+ft_config_headers = [
+  'include/freetype/config/ftmodule.h',
+  'include/freetype/config/ftheader.h',
+  'include/freetype/config/ftstdlib.h',
+  'include/freetype/config/ftconfig.h',
+]
+
+if host_machine.system() == 'windows'
+  winmod = import('windows')
+  base_sources += [
+    'builds/windows/ftdebug.c',
+    winmod.compile_resources('src/base/ftver.rc'),
+  ]
+else
+  base_sources += [
+    'src/base/ftdebug.c',
+  ]
+endif
+
+c_args = [
+  '-DFT2_BUILD_LIBRARY',
+  '-DFT_CONFIG_CONFIG_H=<ftconfig.h>',
+  '-DFT_CONFIG_OPTIONS_H=<ftoption.h>'
+]
+
+if cc.get_id() == 'msvc'
+  c_args += [
+    '-D_CRT_SECURE_NO_WARNINGS',
+    '-D_CRT_NONSTDC_NO_WARNINGS',
+  ]
+endif
+
+check_headers = []
+
+if ['linux', 'darwin', 'cygwin'].contains(host_machine.system())
+  check_headers += [
+    ['unistd.h'],
+    ['fcntl.h'],
+    ['stdint.h'],
+  ]
+  ftconfig_h_in = files('builds/unix/ftconfig.h.in')
+else
+  ftconfig_h_in = files('include/freetype/config/ftconfig.h')
+endif
+
+check_optional_deps = [
+  ['bzip2'],
+  ['harfbuzz', {'version': '>= 1.3.0'}]
+]
+
+conf = configuration_data()
+deps = []
+incbase = include_directories(['include'])
+
+foreach check : check_headers
+  name = check[0]
+
+  if cc.has_header(name)
+    conf.set('HAVE_@0@'.format(name.to_upper().underscorify()), 1)
+  endif
+endforeach
+
+foreach check : check_optional_deps
+  name = check[0]
+  opts = check.length() > 1 ? check[1] : {}
+  pkg_name = opts.get('pkg', name)
+  conf_name = 'FT_CONFIG_OPTION_USE_@0@'.format(name.to_upper())
+
+  if opts.has_key('version')
+    dep = dependency(pkg_name, version : opts['version'], required : false)
+  else
+    dep = dependency(pkg_name, required : false)
+  endif
+
+  conf.set(conf_name, dep.found())
+
+  if dep.found()
+    deps += [dep]
+  endif
+endforeach
+
+zlib_dep = dependency('zlib', required: false, fallback: ['zlib', 'zlib_dep'])
+if zlib_dep.found()
+  conf.set('FT_CONFIG_OPTION_SYSTEM_ZLIB', 1)
+  deps += [zlib_dep]
+endif
+
+png_dep = dependency('libpng',
+  required: false,
+  fallback: ['libpng', 'png_dep']
+)
+if png_dep.found()
+  conf.set('FT_CONFIG_OPTION_USE_PNG', 1)
+  deps += [png_dep]
+endif
+
+configure_file(input: ftconfig_h_in,
+  output: 'ftconfig.h',
+  configuration: conf)
+
+ft_config_headers += [configure_file(input: 'include/freetype/config/ftoption.h.in',
+  output: 'ftoption.h',
+  configuration: conf)]
+
+libfreetype = library('freetype', base_sources,
+  include_directories: incbase,
+  dependencies: deps,
+  c_args: c_args,
+  install: true,
+  version: '6.16.0',
```

Comment:
Where this version is coming from?

Reply:
Same as aboce, here is the CMakeLists equivalent:

https://git.savannah.gnu.org/cgit/freetype/freetype2.git/tree/CMakeLists.txt#n147

Side note: why does the README list `6.16.1`, but the latest CMakeLists still have `6.16.0` ?!

Reply:
Please add a comment saying so. It will help a lot to anyone reading the wrap later.

Reply:
Updated shared object version number eg. 6.16.1 as this is based on release 2.9.1